app/main/lib/text_similarity.py

```diff
@@ -53,8 +85,10 @@ def get_model_and_threshold(search_params):
       model_key = search_params['model']
   if 'threshold' in search_params:
       threshold = search_params['threshold']
-  if 'per_model_threshold' in search_params and search_params['per_model_threshold'].get(model_key):
+  if 'per_model_threshold' in search_params and isinstance(search_params['per_model_threshold'], dict) and search_params['per_model_threshold'].get(model_key):
       threshold = search_params['per_model_threshold'].get(model_key)
+  if 'per_model_threshold' in search_params and isinstance(search_params['per_model_threshold'], list) and [e for e in search_params['per_model_threshold'] if e["model"] == model_key]:
+      threshold = [e for e in search_params['per_model_threshold'] if e["model"] == model_key][0]["value"]
   if threshold is None:
       threshold = 0.9
```

Comment:
Let's please add a warning here that no threshold was supplied and 0.9 is being used as a default.

Reply:
sounds good